cmake/SetupBlaze.cmake

```diff
@@ -19,24 +19,32 @@ set_property(TARGET Blaze PROPERTY
   INTERFACE_INCLUDE_DIRECTORIES ${BLAZE_INCLUDE_DIR})
 set_property(TARGET Blaze PROPERTY
   INTERFACE_LINK_LIBRARIES Lapack)
+target_link_libraries(
+  Blaze
+  INTERFACE
+  Blas
+  GSL::gsl # for BLAS header
+  Lapack
+  )
 
 target_compile_definitions(Blaze
   INTERFACE
-  # Override SMP configurations
+  # Enable external BLAS kernels
+  BLAZE_BLAS_MODE=1
+  # Set default matrix storage order to column-major
+  BLAZE_DEFAULT_STORAGE_ORDER=blaze::columnMajor
+  # Disable SMP parallelization
   BLAZE_USE_SHARED_MEMORY_PARALLELIZATION=0
-  BLAZE_OPENMP_PARALLEL_MODE=0
-  BLAZE_CPP_THREADS_PARALLEL_MODE=0
-  BLAZE_BOOST_THREADS_PARALLEL_MODE=0
   # Disable MPI parallelization
   BLAZE_MPI_PARALLEL_MODE=0
-  # Disable HPX parallelization
-  BLAZE_HPX_PARALLEL_MODE=0
-  BLAZE_USE_PADDING=0
```

Comment:
let's not enable padding... That is likely going to cause headaches and there's no data indicating it'll improve anything.

Reply:
Blaze warns that disabling padding can considerably decrease performance for dense matrix operations: https://bitbucket.org/blaze-lib/blaze/src/c4d9e85414370e880e5e79c86e3c8d4d38dcde7a/blaze/config/Optimizations.h#lines-52

To support padding, all you have to do is use `.spacing()` instead of `.rows()/.columns()` in the `LDA`, `LDB`, etc. parameters of LAPACK.

I split this change into a separate commit.

Reply:
I don't know how this translates to LIBXSMM. Naively I assume it would be similar speedups, but I think it would be good to do a microbenchmark on a machine with AVX or AVX2 support since that's where I expect the impact will be most noticeable. The reason being that unaligned accesses used to be very slow with vectorization but CPUs over the last 4-5 years don't have that issue anymore, but I don't know what the case is with padding. Does that sound okay?